### PR TITLE
fix: catch reinit of replicaset

### PR DESCRIPTION
**Pull request by boredland**
- #278

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -46,3 +46,6 @@ packages/mongodb-memory-server-core/lib
 coverage
 .nyc_output
 package-lock.json
+
+# testfiles
+reuse*
```
Comment (hasezoey) [inline .gitignore]: why do these need to be ignored?

File: packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import MongoMemoryReplSet from '../MongoMemoryReplSet';
+import fs from 'fs';
+import rimraf from 'rimraf';
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 600000;
+
+describe('multi-member replica set', () => {
+  it('should start multiple times', async () => {
+    const testDir = fs.mkdtempSync('reuse_');
```
Comment (hasezoey) [inline packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts]: please refactor to use tmp (package)

File: yarn.lock [hunk 3]
```diff
@@ -1839,6 +1839,14 @@
   resolved "https://registry.yarnpkg.com/@types/node/-/node-12.12.21.tgz#aa44a6363291c7037111c47e4661ad210aded23f"
   integrity sha512-8sRGhbpU+ck1n0PGAUgVrWrWdjSW2aqNeyC15W88GRsMpSwzv6RJGlLhE7s2RhVSOdyDmxbqlWSeThq4/7xqlA==
 
+"@types/rimraf@^2.0.3":
```
Comment (hasezoey) [inline yarn.lock]: i think this should be removed here too

**Comment by hasezoey** [comment]
are you sure you reverted the lockfile correctly?

**Comment by boredland** [comment]
> are you sure you reverted the lockfile correctly?
> -> seems like more is now removed than earlier commits added

I"m sure I reverted that one correctly, but now I just squashed those changes together to cover the unnecessary changes.

**Comment by hasezoey** [comment]
@boredland checked it out locally, and the yarn.lock is still not like the master one (many things removed and set to old versions)

to fix this run `yarn install` in the root of the repo or in the directory the yarn.lock file is in and commit it, you will see if you compare that with nodkz/master that only 3 package versions have changed
-> i would have pushed it to your branch, but it seems like "Allow edits from maintainers" is disabled for this pr / is not possible when you dont own the repo/branch

File: packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+afterEach(() => {
+  tmpDir.removeCallback();
+});
+
+const sleep = (ms: number) => {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
+
+describe('single-member replica set', () => {
+  it('should start multiple times', async () => {
```
Comment (hasezoey) [inline packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts]: maybe an special test-timeout should be set for this test

File: packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import MongoMemoryReplSet from '../MongoMemoryReplSet';
+import * as tmp from 'tmp';
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 600000;
```
Comment (hasezoey) [inline packages/mongodb-memory-server-core/src/__tests__/replset-single-restart-test.ts]: please remove this, otherwise it might affect other tests too

**Comment by sakulstra** [comment]
hayho, what's the status here?

@hasezoey can we somehow help pushing this forward? :pray:

**Comment by nodkz** [comment]
Everything is ok! Bump it!

Tnx 🙏

**Comment by Ranguna** [comment]
Has this been released already ?
I can't seem to find these changes on my locally installed `mongodb-memory-server-core`.

**Comment by hasezoey** [comment]
@Ranguna yes since 6.4.0

**Comment by Ranguna** [comment]
@hasezoey all right, thanks!

**Pull request merged**